### ISSUE TITLE
*ERROR* [CRTC:81:pipe A] mismatch in post_csc_lut hw_state doesn't match sw_state

### ISSUE THREAD

**Issue by makoONE** [bbaa-bbaa/i915-sriov-dkms#10]
Hello everyone,
I use Proxmox on Debian with Xfce and the current kernel versions 6.8.12-4-pve and 6.11.0-1-pve. This causes error entries with the following dmesg output. It works so far and the error seems to have no effect or I haven't noticed it yet. It would still be nice to get rid of it though, any ideas would be appreciated.
```
[   16.455109] i915 0000:00:02.0: [drm] *ERROR* [CRTC:81:pipe A] mismatch in post_csc_lut hw_state doesn't match sw_state
[   16.455115] ------------[ cut here ]------------
[   16.455116] i915 0000:00:02.0: [drm] pipe state doesn't match!
[   16.455168] WARNING: CPU: 6 PID: 1890 at /var/lib/dkms/i915-sriov-dkms-bbaa-bbaa/2024.12.02/build/drivers/gpu/drm/i915/display/intel_modeset_verify.c:222 intel_modeset_verify_crtc+0x5d6/0x6b0 [i915]
[   16.455276] Modules linked in: ebtable_filter ebtables ip_set ip6table_raw iptable_raw ip6table_filter ip6_tables iptable_filter nf_tables qrtr softdog sunrpc binfmt_misc bonding tls nfnetlink_log nfnetlink input_leds joydev hid_apple hid_generic usbkbd usbmouse usbhid apple_mfi_fastcharge hid intel_rapl_msr intel_rapl_common intel_uncore_frequency intel_uncore_frequency_common snd_sof_pci_intel_tgl snd_sof_pci_intel_cnl snd_sof_intel_hda_generic soundwire_intel soundwire_cadence snd_sof_intel_hda_common x86_pkg_temp_thermal snd_sof_intel_hda_mlink intel_powerclamp snd_hda_codec_hdmi snd_sof_intel_hda snd_sof_pci snd_sof_xtensa_dsp snd_sof snd_sof_utils snd_soc_hdac_hda snd_soc_acpi_intel_match soundwire_generic_allocation snd_soc_acpi soundwire_bus coretemp snd_hda_codec_realtek snd_soc_avs kvm_intel snd_soc_hda_codec snd_hda_codec_generic snd_hda_ext_core snd_hda_scodec_component snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine kvm mt7921e mt7921_common crct10dif_pclmul mt792x_lib polyval_clmulni polyval_generic
[   16.455300]  ghash_clmulni_intel mt76_connac_lib sha256_ssse3 sha1_ssse3 mt76 aesni_intel snd_hda_intel snd_intel_dspcfg crypto_simd snd_intel_sdw_acpi mac80211 cryptd snd_hda_codec snd_hda_core btusb btrtl snd_hwdep btintel snd_pcm btbcm btmtk snd_timer rapl cfg80211 bluetooth cmdlinepart spi_nor snd libarc4 intel_cstate mei_me ee1004 soundcore pcspkr mtd wmi_bmof mei intel_pmc_core mei_vsc_hw intel_vsec pmt_telemetry pmt_class acpi_tad acpi_pad mac_hid zfs(PO) spl(O) vhost_net vhost vhost_iotlb tap parport_pc ppdev lp parport efi_pstore dmi_sysfs ip_tables x_tables autofs4 btrfs blake2b_generic xor raid6_pq dm_thin_pool dm_persistent_data dm_bio_prison dm_bufio libcrc32c i915(OE) drm_buddy i2c_algo_bit xhci_pci ttm nvme xhci_pci_renesas drm_display_helper crc32_pclmul ahci nvme_core i2c_i801 xhci_hcd cec spi_intel_pci i2c_mux igc spi_intel libahci i2c_smbus nvme_auth rc_core video wmi
[   16.455333] CPU: 6 UID: 0 PID: 1890 Comm: Xorg Tainted: P     U     OE      6.11.0-1-pve #1
[   16.455335] Tainted: [P]=PROPRIETARY_MODULE, [U]=USER, [O]=OOT_MODULE, [E]=UNSIGNED_MODULE
[   16.455336] Hardware name: Micro Computer(HK) Tech Limited Venus series/AHBNB, BIOS 1.0S 05/11/2024
[   16.455337] RIP: 0010:intel_modeset_verify_crtc+0x5d6/0x6b0 [i915]
[   16.455413] Code: fc ff ff 49 8b 7f 08 48 8b 5f 50 48 85 db 75 03 48 8b 1f e8 1c 02 ec cd 48 89 da 48 c7 c7 18 9c 89 c0 48 89 c6 e8 ca 54 49 cd <0f> 0b e9 89 fd ff ff 80 fb 01 0f 87 ae f7 0c 00 41 0f b6 8c 24 50
[   16.455414] RSP: 0018:ffffa84602183868 EFLAGS: 00010246
[   16.455416] RAX: 0000000000000000 RBX: ffff8b01826e1e90 RCX: 0000000000000000
[   16.455416] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000000
[   16.455417] RBP: ffffa846021838e0 R08: 0000000000000000 R09: 0000000000000000
[   16.455417] R10: 0000000000000000 R11: 0000000000000000 R12: ffff8b018b2fe000
[   16.455418] R13: ffff8b018b2fc000 R14: ffff8b01802d6000 R15: ffff8b01ac628000
[   16.455418] FS:  00007c6b4ae16ac0(0000) GS:ffff8b10ef700000(0000) knlGS:0000000000000000
[   16.455419] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   16.455420] CR2: 00007c6b336fc000 CR3: 00000001161b8000 CR4: 0000000000f50ef0
[   16.455421] PKRU: 55555554
[   16.455421] Call Trace:
[   16.455422]  <TASK>
[   16.455424]  ? show_regs+0x6c/0x80
[   16.455428]  ? __warn+0x88/0x140
[   16.455429]  ? intel_modeset_verify_crtc+0x5d6/0x6b0 [i915]
[   16.455503]  ? report_bug+0x182/0x1b0
[   16.455506]  ? handle_bug+0x46/0x90
[   16.455508]  ? exc_invalid_op+0x18/0x80
[   16.455508]  ? asm_exc_invalid_op+0x1b/0x20
[   16.455511]  ? intel_modeset_verify_crtc+0x5d6/0x6b0 [i915]
[   16.455582]  ? intel_modeset_verify_crtc+0x5d6/0x6b0 [i915]
[   16.455651]  ? intel_fbc_nuke+0x42/0xd0 [i915]
[   16.455725]  intel_atomic_commit_tail+0x974/0xf60 [i915]
[   16.455806]  intel_atomic_commit+0x3bd/0x410 [i915]
[   16.455885]  drm_atomic_commit+0xa5/0xe0
[   16.455888]  ? __pfx___drm_printfn_info+0x10/0x10
[   16.455891]  drm_atomic_helper_set_config+0x82/0xd0
[   16.455893]  drm_mode_setcrtc+0x3f4/0x990
[   16.455895]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[   16.455897]  drm_ioctl_kernel+0xb9/0x120
[   16.455899]  drm_ioctl+0x2f3/0x580
[   16.455900]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[   16.455902]  __x64_sys_ioctl+0xa0/0xf0
[   16.455904]  x64_sys_call+0xb31/0x24e0
[   16.455906]  do_syscall_64+0x7e/0x170
[   16.455908]  ? do_fault+0x179/0x490
[   16.455910]  ? __handle_mm_fault+0x83f/0x1120
[   16.455912]  ? __count_memcg_events+0x7d/0x130
[   16.455914]  ? __rseq_handle_notify_resume+0xa4/0x4e0
[   16.455916]  ? count_memcg_events.constprop.0+0x2a/0x50
[   16.455918]  ? irqentry_exit_to_user_mode+0x182/0x250
[   16.455920]  ? irqentry_exit+0x43/0x50
[   16.455922]  ? exc_page_fault+0x96/0x1e0
[   16.455923]  entry_SYSCALL_64_after_hwframe+0x76/0x7e
[   16.455924] RIP: 0033:0x7c6b4b11ccdb
[   16.455926] Code: 00 48 89 44 24 18 31 c0 48 8d 44 24 60 c7 04 24 10 00 00 00 48 89 44 24 08 48 8d 44 24 20 48 89 44 24 10 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 1c 48 8b 44 24 18 64 48 2b 04 25 28 00 00
[   16.455927] RSP: 002b:00007fffc2d3d540 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[   16.455928] RAX: ffffffffffffffda RBX: 000061654705b4d0 RCX: 00007c6b4b11ccdb
[   16.455928] RDX: 00007fffc2d3d670 RSI: 00000000c06864a2 RDI: 000000000000000d
[   16.455929] RBP: 00007fffc2d3d670 R08: 000000000004100a R09: 00000001012fc000
[   16.455929] R10: 000061654705b170 R11: 0000000000000246 R12: 00000000c06864a2
[   16.455930] R13: 000000000000000d R14: 000061654705b260 R15: 0000000000000001
[   16.455931]  </TASK>
[   16.455931] ---[ end trace 0000000000000000 ]---

```

**Comment by bbaa-bbaa** [comment]
Is your computer connected to a physical display?
Could you add `drm.debug=0x19f log_buf_len=4m` to the kernel command line to export the related DRM kernel logs?

**Comment by makoONE** [comment]
Yes, the computer, my Proxmox server, is connected to a monitor. The error appears every time I start the graphical desktop environment with the startx command. As I wrote, the Xfce interface works so far, but I have noticed that certain console applications in a window, such as Htop or Midnight Commander, cause the entire GUI to freeze when their windows are enlarged. This may be related to the error message. However, the operating system itself is still functional afterwards.
A Windows 11 VM running in parallel on the server with iGPU shared via SR-IOV always runs flawlessly.

Briefly about the background of my setup: My Proxmox server usually runs some typical home server applications without the graphical user interface being activated. Occasionally I want to use the server additionally as a (spare) workstation, for this I start the X-GUI and can then connect with Remmina/RDP against my Windows VM running on the same server. 

I have implemented your suggested options, here are the logs after starting the GUI with startx.
[logs.zip](https://github.com/user-attachments/files/18052208/logs.zip)

**Comment by bbaa-bbaa** [comment]
> I have implemented your suggested options, here are the logs after starting the GUI with startx.

It seems that the 4MB kernel dmesg buffer is far from sufficient for DRM logs. Could you try using journalctl -k instead of dmesg to export more detailed logs?

**Comment by makoONE** [comment]
Sure, here it is.
[journalctl.zip](https://github.com/user-attachments/files/18052280/journalctl.zip)

**Comment by bbaa-bbaa** [comment]
Thanks! Remember to remove drm.debug=0x19f and log_buf_len=4m in a timely manner, or the rapidly growing logs may eat up your hard drive space.
 You can use echo 0 | tee /sys/module/drm/parameters/debug to disable DRM debug without restarting the computer.

**Comment by patrykk** [comment]
I have that same problem:

`[ 8846.074437] i915 0000:00:02.0: [drm] *ERROR* [CRTC:132:pipe B] mismatch in post_csc_lut hw_state doesn't match sw_state
[ 8846.074453] ------------[ cut here ]------------
[ 8846.074455] i915 0000:00:02.0: [drm] pipe state doesn't match!
[ 8846.074621] WARNING: CPU: 6 PID: 2514 at /var/lib/dkms/i915-sriov-dkms/2024.12.02/build/drivers/gpu/drm/i915/display/intel_modeset_verify.c:222 intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.074827] Modules linked in: tls hid_multitouch nft_masq nft_ct xt_conntrack xt_MASQUERADE nft_reject_ipv4 nf_conntrack_netlink nf_reject_ipv4 nft_reject xfrm_user xfrm_algo act_csum cls_u32 sch_htb xt_addrtype nft_compat nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 scsi_transport_iscsi overlay cmac algif_hash algif_skcipher af_alg nf_tables usbip_host pkcs8_key_parser snd_seq_dummy snd_seq_oss snd_seq_midi_event snd_seq cuse elan_i2c ec_sys v4l2loopback(OE) kvmgt mdev vfio_iommu_type1 vfio iommufd vhost_net vhost vhost_iotlb tap acpi_power_meter uvcvideo videobuf2_vmalloc uvc videobuf2_memops videobuf2_v4l2 videobuf2_common videodev usbhid sunrpc acpi_ipmi virtio_mmio virtio_scsi virtio_rng 9pnet_virtio 9pnet netfs virtio_console virtio_net net_failover failover kvm_intel kvm ipmi_watchdog ipmi_devintf ipmi_msghandler ums_cypress lis3lv02d_i2c lis3lv02d btsdio btusb btmtk firewire_ohci firewire_core crc_itu_t dcdbas dell_wmi_descriptor dm_zero coretemp mspro_block tifm_ms memstick tifm_sd
[ 8846.074886]  mmc_core tifm_core snd_soc_skl_hda_dsp uio_cif snd_soc_intel_sof_board_helpers uio snd_soc_intel_hda_dsp_common hidp snd_sof_probes bnep vfat tun fat snd_hda_codec_hdmi rfcomm snd_hda_codec_conexant snd_hda_codec_generic snd_soc_dmic snd_usb_audio snd_usbmidi_lib snd_ump snd_sof_pci_intel_mtl snd_rawmidi snd_sof_intel_hda_generic snd_seq_device soundwire_intel mc soundwire_cadence snd_sof_intel_hda_common hci_uart snd_soc_hdac_hda btqca pwrseq_core snd_sof_intel_hda_mlink btrtl snd_sof_intel_hda btintel snd_sof_pci snd_sof_xtensa_dsp btbcm snd_sof snd_sof_utils snd_hda_ext_core snd_soc_acpi_intel_match soundwire_generic_allocation snd_soc_acpi soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel processor_thermal_device_pci mousedev bluetooth processor_thermal_device snd_intel_dspcfg psmouse serio_raw processor_thermal_wt_hint snd_intel_sdw_acpi mei_gsc_proxy spd5118 spi_nor processor_thermal_rfim atkbd think_lmi snd_hda_codec libps2 intel_rapl_msr vivaldi_fmap snd_ctl_led mtd
[ 8846.074937]  firmware_attributes_class mei_me i2c_i801 snd_hda_core snd_pcm_oss processor_thermal_rapl intel_lpss_pci i2c_smbus wmi_bmof snd_hwdep snd_mixer_oss ucsi_acpi intel_rapl_common intel_lpss i2c_mux processor_thermal_wt_req mei typec_ucsi thunderbolt snd_pcm processor_thermal_power_floor thinkpad_acpi intel_vpu idma64 typec processor_thermal_mbox igen6_edac snd_timer platform_profile roles i2c_hid_acpi snd joydev intel_pmc_core soundcore i2c_hid int3403_thermal yenta_socket int340x_thermal_zone pcmcia_rsrc intel_vsec i8042 intel_hid pcmcia_core pmt_telemetry int3400_thermal acpi_tad serio mac_hid pmt_class pinctrl_meteorlake sparse_keymap acpi_pad acpi_thermal_rel iwlmvm mac80211 e1000e zram libarc4 842_decompress ptp 842_compress lz4hc_compress pps_core lz4_compress hid_generic iwlwifi cfg80211 rfkill uas usb_storage exfat vhci_hcd usbip_core serpent_avx_x86_64 twofish_avx_x86_64 twofish_x86_64_3way twofish_x86_64 twofish_generic twofish_common serpent_generic ccm xts ntfs3 br_netfilter bridge stp llc uinput
[ 8846.074998]  dell_smm_hwmon i2c_dev sg vhba(OE) crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables dm_crypt cbc encrypted_keys trusted asn1_encoder tee crct10dif_pclmul crc32_pclmul polyval_clmulni polyval_generic ghash_clmulni_intel sha512_ssse3 sha256_ssse3 sha1_ssse3 aesni_intel gf128mul nvme crypto_simd cryptd spi_intel_pci nvme_core spi_intel nvme_auth raid1 dm_raid raid456 libcrc32c async_raid6_recov async_memcpy async_pq async_xor xor async_tx raid6_pq dm_mod md_mod i915(OE) drm_buddy drm_display_helper ttm video wmi i2c_algo_bit cec intel_gtt ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2 [last unloaded: hid_multitouch]
[ 8846.075044] CPU: 6 UID: 0 PID: 2514 Comm: Xorg Tainted: G     U  W  OE      6.12.4-arch1-1.1 #1 606ae73903c461ea514adc42e53e980d964fe09f
[ 8846.075050] Tainted: [U]=USER, [W]=WARN, [O]=OOT_MODULE, [E]=UNSIGNED_MODULE
[ 8846.075052] Hardware name: LENOVO 21M7002VPB/21M7002VPB, BIOS R2JET35W(1.12 ) 07/05/2024
[ 8846.075056] RIP: 0010:intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.075211] Code: b3 0b 00 49 8b 7d 08 48 8b 5f 50 48 85 db 75 03 48 8b 1f e8 5c ef bb c1 48 89 da 48 c7 c7 18 d1 bc c0 48 89 c6 e8 aa 36 13 c1 <0f> 0b 31 f6 4c 89 e7 48 c7 c2 e3 24 bf c0 e8 27 2a fb ff 48 c7 c2
[ 8846.075213] RSP: 0018:ffffae6b013e3810 EFLAGS: 00010286
[ 8846.075217] RAX: 0000000000000000 RBX: ffff9c5d025467a0 RCX: 0000000000000027
[ 8846.075219] RDX: ffff9c605f7218c8 RSI: 0000000000000001 RDI: ffff9c605f7218c0
[ 8846.075221] RBP: ffff9c5fa9494000 R08: 00000000ffff7fff R09: ffff9c606fe3ffa8
[ 8846.075223] R10: 00000000ffff7fff R11: 0000000000000003 R12: ffff9c5fa9496000
[ 8846.075224] R13: ffff9c5d09c50000 R14: ffff9c5d03b19000 R15: ffff9c5d09c502b0
[ 8846.075226] FS:  00007ab87b7a49c0(0000) GS:ffff9c605f700000(0000) knlGS:0000000000000000
[ 8846.075229] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 8846.075230] CR2: 000077fe00232000 CR3: 0000000166578001 CR4: 0000000000f72ef0
[ 8846.075232] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 8846.075234] DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
[ 8846.075235] PKRU: 55555554
[ 8846.075237] Call Trace:
[ 8846.075242]  <TASK>
[ 8846.075244]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.075418]  ? __warn.cold+0x93/0xed
[ 8846.075428]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.075580]  ? report_bug+0xe7/0x210
[ 8846.075584]  ? handle_bug+0x58/0x90
[ 8846.075589]  ? exc_invalid_op+0x19/0xc0
[ 8846.075592]  ? asm_exc_invalid_op+0x1a/0x20
[ 8846.075597]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.075749]  ? intel_fbc_activate+0x2e/0x90 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.075912]  intel_atomic_commit_tail+0x8e1/0xe70 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.076092]  intel_atomic_commit+0x33e/0x380 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.076263]  drm_atomic_commit+0xa9/0xe0
[ 8846.076269]  ? __pfx___drm_printfn_info+0x10/0x10
[ 8846.076274]  drm_atomic_helper_set_config+0x74/0xb0
[ 8846.076279]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[ 8846.076283]  drm_mode_setcrtc+0x248/0x9d0
[ 8846.076286]  ? drm_mode_object_put+0x2d/0x70
[ 8846.076291]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[ 8846.076293]  drm_ioctl_kernel+0xb0/0x100
[ 8846.076296]  drm_ioctl+0x277/0x4c0
[ 8846.076299]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[ 8846.076302]  __x64_sys_ioctl+0x94/0xd0
[ 8846.076307]  do_syscall_64+0x82/0x190
[ 8846.076311]  ? __pfx_drm_mode_getconnector+0x10/0x10
[ 8846.076313]  ? drm_ioctl_kernel+0xb0/0x100
[ 8846.076315]  ? __check_object_size+0x50/0x210
[ 8846.076319]  ? drm_ioctl+0x2a1/0x4c0
[ 8846.076322]  ? __pfx_drm_mode_getconnector+0x10/0x10
[ 8846.076325]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.076330]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.076333]  ? do_syscall_64+0x8e/0x190
[ 8846.076335]  ? do_syscall_64+0x8e/0x190
[ 8846.076338]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.076341]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.076344]  ? do_syscall_64+0x8e/0x190
[ 8846.076346]  ? do_syscall_64+0x8e/0x190
[ 8846.076348]  ? do_syscall_64+0x8e/0x190
[ 8846.076351]  entry_SYSCALL_64_after_hwframe+0x76/0x7e
[ 8846.076355] RIP: 0033:0x7ab87c025ced
[ 8846.076448] Code: 04 25 28 00 00 00 48 89 45 c8 31 c0 48 8d 45 10 c7 45 b0 10 00 00 00 48 89 45 b8 48 8d 45 d0 48 89 45 c0 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 1a 48 8b 45 c8 64 48 2b 04 25 28 00 00 00
[ 8846.076451] RSP: 002b:00007ffcb019ce60 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 8846.076454] RAX: ffffffffffffffda RBX: 00005804ebe21ab0 RCX: 00007ab87c025ced
[ 8846.076456] RDX: 00007ffcb019cef0 RSI: 00000000c06864a2 RDI: 0000000000000010
[ 8846.076458] RBP: 00007ffcb019ceb0 R08: 0000000000000000 R09: 0000000000000000
[ 8846.076459] R10: 00005804ebc467e8 R11: 0000000000000246 R12: 00007ffcb019cef0
[ 8846.076461] R13: 00000000c06864a2 R14: 0000000000000010 R15: 0000000000000001
[ 8846.076464]  </TASK>
[ 8846.076466] ---[ end trace 0000000000000000 ]---
[ 8846.108413] i915 0000:00:02.0: [drm] *ERROR* [CRTC:132:pipe B] mismatch in post_csc_lut hw_state doesn't match sw_state
[ 8846.108428] ------------[ cut here ]------------
[ 8846.108431] i915 0000:00:02.0: [drm] pipe state doesn't match!
[ 8846.108603] WARNING: CPU: 3 PID: 2514 at /var/lib/dkms/i915-sriov-dkms/2024.12.02/build/drivers/gpu/drm/i915/display/intel_modeset_verify.c:222 intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.108896] Modules linked in: tls hid_multitouch nft_masq nft_ct xt_conntrack xt_MASQUERADE nft_reject_ipv4 nf_conntrack_netlink nf_reject_ipv4 nft_reject xfrm_user xfrm_algo act_csum cls_u32 sch_htb xt_addrtype nft_compat nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 scsi_transport_iscsi overlay cmac algif_hash algif_skcipher af_alg nf_tables usbip_host pkcs8_key_parser snd_seq_dummy snd_seq_oss snd_seq_midi_event snd_seq cuse elan_i2c ec_sys v4l2loopback(OE) kvmgt mdev vfio_iommu_type1 vfio iommufd vhost_net vhost vhost_iotlb tap acpi_power_meter uvcvideo videobuf2_vmalloc uvc videobuf2_memops videobuf2_v4l2 videobuf2_common videodev usbhid sunrpc acpi_ipmi virtio_mmio virtio_scsi virtio_rng 9pnet_virtio 9pnet netfs virtio_console virtio_net net_failover failover kvm_intel kvm ipmi_watchdog ipmi_devintf ipmi_msghandler ums_cypress lis3lv02d_i2c lis3lv02d btsdio btusb btmtk firewire_ohci firewire_core crc_itu_t dcdbas dell_wmi_descriptor dm_zero coretemp mspro_block tifm_ms memstick tifm_sd
[ 8846.108967]  mmc_core tifm_core snd_soc_skl_hda_dsp uio_cif snd_soc_intel_sof_board_helpers uio snd_soc_intel_hda_dsp_common hidp snd_sof_probes bnep vfat tun fat snd_hda_codec_hdmi rfcomm snd_hda_codec_conexant snd_hda_codec_generic snd_soc_dmic snd_usb_audio snd_usbmidi_lib snd_ump snd_sof_pci_intel_mtl snd_rawmidi snd_sof_intel_hda_generic snd_seq_device soundwire_intel mc soundwire_cadence snd_sof_intel_hda_common hci_uart snd_soc_hdac_hda btqca pwrseq_core snd_sof_intel_hda_mlink btrtl snd_sof_intel_hda btintel snd_sof_pci snd_sof_xtensa_dsp btbcm snd_sof snd_sof_utils snd_hda_ext_core snd_soc_acpi_intel_match soundwire_generic_allocation snd_soc_acpi soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel processor_thermal_device_pci mousedev bluetooth processor_thermal_device snd_intel_dspcfg psmouse serio_raw processor_thermal_wt_hint snd_intel_sdw_acpi mei_gsc_proxy spd5118 spi_nor processor_thermal_rfim atkbd think_lmi snd_hda_codec libps2 intel_rapl_msr vivaldi_fmap snd_ctl_led mtd
[ 8846.109030]  firmware_attributes_class mei_me i2c_i801 snd_hda_core snd_pcm_oss processor_thermal_rapl intel_lpss_pci i2c_smbus wmi_bmof snd_hwdep snd_mixer_oss ucsi_acpi intel_rapl_common intel_lpss i2c_mux processor_thermal_wt_req mei typec_ucsi thunderbolt snd_pcm processor_thermal_power_floor thinkpad_acpi intel_vpu idma64 typec processor_thermal_mbox igen6_edac snd_timer platform_profile roles i2c_hid_acpi snd joydev intel_pmc_core soundcore i2c_hid int3403_thermal yenta_socket int340x_thermal_zone pcmcia_rsrc intel_vsec i8042 intel_hid pcmcia_core pmt_telemetry int3400_thermal acpi_tad serio mac_hid pmt_class pinctrl_meteorlake sparse_keymap acpi_pad acpi_thermal_rel iwlmvm mac80211 e1000e zram libarc4 842_decompress ptp 842_compress lz4hc_compress pps_core lz4_compress hid_generic iwlwifi cfg80211 rfkill uas usb_storage exfat vhci_hcd usbip_core serpent_avx_x86_64 twofish_avx_x86_64 twofish_x86_64_3way twofish_x86_64 twofish_generic twofish_common serpent_generic ccm xts ntfs3 br_netfilter bridge stp llc uinput
[ 8846.109101]  dell_smm_hwmon i2c_dev sg vhba(OE) crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables dm_crypt cbc encrypted_keys trusted asn1_encoder tee crct10dif_pclmul crc32_pclmul polyval_clmulni polyval_generic ghash_clmulni_intel sha512_ssse3 sha256_ssse3 sha1_ssse3 aesni_intel gf128mul nvme crypto_simd cryptd spi_intel_pci nvme_core spi_intel nvme_auth raid1 dm_raid raid456 libcrc32c async_raid6_recov async_memcpy async_pq async_xor xor async_tx raid6_pq dm_mod md_mod i915(OE) drm_buddy drm_display_helper ttm video wmi i2c_algo_bit cec intel_gtt ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2 [last unloaded: hid_multitouch]
[ 8846.109157] CPU: 3 UID: 0 PID: 2514 Comm: Xorg Tainted: G     U  W  OE      6.12.4-arch1-1.1 #1 606ae73903c461ea514adc42e53e980d964fe09f
[ 8846.109165] Tainted: [U]=USER, [W]=WARN, [O]=OOT_MODULE, [E]=UNSIGNED_MODULE
[ 8846.109167] Hardware name: LENOVO 21M7002VPB/21M7002VPB, BIOS R2JET35W(1.12 ) 07/05/2024
[ 8846.109170] RIP: 0010:intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.109361] Code: b3 0b 00 49 8b 7d 08 48 8b 5f 50 48 85 db 75 03 48 8b 1f e8 5c ef bb c1 48 89 da 48 c7 c7 18 d1 bc c0 48 89 c6 e8 aa 36 13 c1 <0f> 0b 31 f6 4c 89 e7 48 c7 c2 e3 24 bf c0 e8 27 2a fb ff 48 c7 c2
[ 8846.109364] RSP: 0018:ffffae6b013e3910 EFLAGS: 00010286
[ 8846.109368] RAX: 0000000000000000 RBX: ffff9c5d025467a0 RCX: 0000000000000027
[ 8846.109372] RDX: ffff9c605f5a18c8 RSI: 0000000000000001 RDI: ffff9c605f5a18c0
[ 8846.109374] RBP: ffff9c6048a72000 R08: 00000000ffff7fff R09: ffff9c606fe3ffa8
[ 8846.109376] R10: 00000000ffff7fff R11: 0000000000000003 R12: ffff9c5f0608e000
[ 8846.109377] R13: ffff9c5d09c50000 R14: ffff9c5d03b19000 R15: ffff9c5d09c502b0
[ 8846.109379] FS:  00007ab87b7a49c0(0000) GS:ffff9c605f580000(0000) knlGS:0000000000000000
[ 8846.109382] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 8846.109384] CR2: 00007d4a6a144010 CR3: 0000000166578002 CR4: 0000000000f72ef0
[ 8846.109386] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 8846.109388] DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
[ 8846.109390] PKRU: 55555554
[ 8846.109392] Call Trace:
[ 8846.109397]  <TASK>
[ 8846.109399]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.109585]  ? __warn.cold+0x93/0xed
[ 8846.109593]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.109778]  ? report_bug+0xe7/0x210
[ 8846.109783]  ? handle_bug+0x58/0x90
[ 8846.109790]  ? exc_invalid_op+0x19/0xc0
[ 8846.109794]  ? asm_exc_invalid_op+0x1a/0x20
[ 8846.109800]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.109979]  ? intel_fbc_activate+0x2e/0x90 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.110165]  intel_atomic_commit_tail+0x8e1/0xe70 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.110371]  intel_atomic_commit+0x33e/0x380 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.110572]  drm_atomic_commit+0xa9/0xe0
[ 8846.110577]  ? __pfx___drm_printfn_info+0x10/0x10
[ 8846.110581]  drm_atomic_helper_set_config+0x74/0xb0
[ 8846.110585]  drm_mode_setcrtc+0x248/0x9d0
[ 8846.110589]  ? __pfx_drm_mode_obj_set_property_ioctl+0x10/0x10
[ 8846.110594]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[ 8846.110597]  drm_ioctl_kernel+0xb0/0x100
[ 8846.110600]  drm_ioctl+0x277/0x4c0
[ 8846.110603]  ? __pfx_drm_mode_setcrtc+0x10/0x10
[ 8846.110607]  __x64_sys_ioctl+0x94/0xd0
[ 8846.110611]  do_syscall_64+0x82/0x190
[ 8846.110614]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.110619]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.110623]  ? do_syscall_64+0x8e/0x190
[ 8846.110626]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.110629]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.110633]  ? do_syscall_64+0x8e/0x190
[ 8846.110635]  ? do_syscall_64+0x8e/0x190
[ 8846.110637]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.110641]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.110645]  ? do_syscall_64+0x8e/0x190
[ 8846.110647]  ? do_syscall_64+0x8e/0x190
[ 8846.110649]  ? do_syscall_64+0x8e/0x190
[ 8846.110652]  entry_SYSCALL_64_after_hwframe+0x76/0x7e
[ 8846.110655] RIP: 0033:0x7ab87c025ced
[ 8846.110691] Code: 04 25 28 00 00 00 48 89 45 c8 31 c0 48 8d 45 10 c7 45 b0 10 00 00 00 48 89 45 b8 48 8d 45 d0 48 89 45 c0 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 1a 48 8b 45 c8 64 48 2b 04 25 28 00 00 00
[ 8846.110693] RSP: 002b:00007ffcb019cc90 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 8846.110697] RAX: ffffffffffffffda RBX: 00005804ebe230b0 RCX: 00007ab87c025ced
[ 8846.110699] RDX: 00007ffcb019cd20 RSI: 00000000c06864a2 RDI: 0000000000000010
[ 8846.110701] RBP: 00007ffcb019cce0 R08: 0000000000000000 R09: 00005804ecb76380
[ 8846.110702] R10: 00005804ebe1e258 R11: 0000000000000246 R12: 00007ffcb019cd20
[ 8846.110704] R13: 00000000c06864a2 R14: 0000000000000010 R15: 0000000000000001
[ 8846.110708]  </TASK>
[ 8846.110709] ---[ end trace 0000000000000000 ]---
[ 8846.141750] i915 0000:00:02.0: [drm] *ERROR* [CRTC:132:pipe B] mismatch in post_csc_lut hw_state doesn't match sw_state
[ 8846.141767] ------------[ cut here ]------------
[ 8846.141770] i915 0000:00:02.0: [drm] pipe state doesn't match!
[ 8846.141984] WARNING: CPU: 3 PID: 2514 at /var/lib/dkms/i915-sriov-dkms/2024.12.02/build/drivers/gpu/drm/i915/display/intel_modeset_verify.c:222 intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.142222] Modules linked in: tls hid_multitouch nft_masq nft_ct xt_conntrack xt_MASQUERADE nft_reject_ipv4 nf_conntrack_netlink nf_reject_ipv4 nft_reject xfrm_user xfrm_algo act_csum cls_u32 sch_htb xt_addrtype nft_compat nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 scsi_transport_iscsi overlay cmac algif_hash algif_skcipher af_alg nf_tables usbip_host pkcs8_key_parser snd_seq_dummy snd_seq_oss snd_seq_midi_event snd_seq cuse elan_i2c ec_sys v4l2loopback(OE) kvmgt mdev vfio_iommu_type1 vfio iommufd vhost_net vhost vhost_iotlb tap acpi_power_meter uvcvideo videobuf2_vmalloc uvc videobuf2_memops videobuf2_v4l2 videobuf2_common videodev usbhid sunrpc acpi_ipmi virtio_mmio virtio_scsi virtio_rng 9pnet_virtio 9pnet netfs virtio_console virtio_net net_failover failover kvm_intel kvm ipmi_watchdog ipmi_devintf ipmi_msghandler ums_cypress lis3lv02d_i2c lis3lv02d btsdio btusb btmtk firewire_ohci firewire_core crc_itu_t dcdbas dell_wmi_descriptor dm_zero coretemp mspro_block tifm_ms memstick tifm_sd
[ 8846.142288]  mmc_core tifm_core snd_soc_skl_hda_dsp uio_cif snd_soc_intel_sof_board_helpers uio snd_soc_intel_hda_dsp_common hidp snd_sof_probes bnep vfat tun fat snd_hda_codec_hdmi rfcomm snd_hda_codec_conexant snd_hda_codec_generic snd_soc_dmic snd_usb_audio snd_usbmidi_lib snd_ump snd_sof_pci_intel_mtl snd_rawmidi snd_sof_intel_hda_generic snd_seq_device soundwire_intel mc soundwire_cadence snd_sof_intel_hda_common hci_uart snd_soc_hdac_hda btqca pwrseq_core snd_sof_intel_hda_mlink btrtl snd_sof_intel_hda btintel snd_sof_pci snd_sof_xtensa_dsp btbcm snd_sof snd_sof_utils snd_hda_ext_core snd_soc_acpi_intel_match soundwire_generic_allocation snd_soc_acpi soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel processor_thermal_device_pci mousedev bluetooth processor_thermal_device snd_intel_dspcfg psmouse serio_raw processor_thermal_wt_hint snd_intel_sdw_acpi mei_gsc_proxy spd5118 spi_nor processor_thermal_rfim atkbd think_lmi snd_hda_codec libps2 intel_rapl_msr vivaldi_fmap snd_ctl_led mtd
[ 8846.142342]  firmware_attributes_class mei_me i2c_i801 snd_hda_core snd_pcm_oss processor_thermal_rapl intel_lpss_pci i2c_smbus wmi_bmof snd_hwdep snd_mixer_oss ucsi_acpi intel_rapl_common intel_lpss i2c_mux processor_thermal_wt_req mei typec_ucsi thunderbolt snd_pcm processor_thermal_power_floor thinkpad_acpi intel_vpu idma64 typec processor_thermal_mbox igen6_edac snd_timer platform_profile roles i2c_hid_acpi snd joydev intel_pmc_core soundcore i2c_hid int3403_thermal yenta_socket int340x_thermal_zone pcmcia_rsrc intel_vsec i8042 intel_hid pcmcia_core pmt_telemetry int3400_thermal acpi_tad serio mac_hid pmt_class pinctrl_meteorlake sparse_keymap acpi_pad acpi_thermal_rel iwlmvm mac80211 e1000e zram libarc4 842_decompress ptp 842_compress lz4hc_compress pps_core lz4_compress hid_generic iwlwifi cfg80211 rfkill uas usb_storage exfat vhci_hcd usbip_core serpent_avx_x86_64 twofish_avx_x86_64 twofish_x86_64_3way twofish_x86_64 twofish_generic twofish_common serpent_generic ccm xts ntfs3 br_netfilter bridge stp llc uinput
[ 8846.142407]  dell_smm_hwmon i2c_dev sg vhba(OE) crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables dm_crypt cbc encrypted_keys trusted asn1_encoder tee crct10dif_pclmul crc32_pclmul polyval_clmulni polyval_generic ghash_clmulni_intel sha512_ssse3 sha256_ssse3 sha1_ssse3 aesni_intel gf128mul nvme crypto_simd cryptd spi_intel_pci nvme_core spi_intel nvme_auth raid1 dm_raid raid456 libcrc32c async_raid6_recov async_memcpy async_pq async_xor xor async_tx raid6_pq dm_mod md_mod i915(OE) drm_buddy drm_display_helper ttm video wmi i2c_algo_bit cec intel_gtt ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2 [last unloaded: hid_multitouch]
[ 8846.142458] CPU: 3 UID: 0 PID: 2514 Comm: Xorg Tainted: G     U  W  OE      6.12.4-arch1-1.1 #1 606ae73903c461ea514adc42e53e980d964fe09f
[ 8846.142465] Tainted: [U]=USER, [W]=WARN, [O]=OOT_MODULE, [E]=UNSIGNED_MODULE
[ 8846.142467] Hardware name: LENOVO 21M7002VPB/21M7002VPB, BIOS R2JET35W(1.12 ) 07/05/2024
[ 8846.142470] RIP: 0010:intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.142645] Code: b3 0b 00 49 8b 7d 08 48 8b 5f 50 48 85 db 75 03 48 8b 1f e8 5c ef bb c1 48 89 da 48 c7 c7 18 d1 bc c0 48 89 c6 e8 aa 36 13 c1 <0f> 0b 31 f6 4c 89 e7 48 c7 c2 e3 24 bf c0 e8 27 2a fb ff 48 c7 c2
[ 8846.142648] RSP: 0018:ffffae6b013e3780 EFLAGS: 00010286
[ 8846.142652] RAX: 0000000000000000 RBX: ffff9c5d025467a0 RCX: 0000000000000027
[ 8846.142655] RDX: ffff9c605f5a18c8 RSI: 0000000000000001 RDI: ffff9c605f5a18c0
[ 8846.142657] RBP: ffff9c6048a72000 R08: 00000000ffff7fff R09: ffff9c606fe3ffa8
[ 8846.142659] R10: 00000000ffff7fff R11: 0000000000000003 R12: ffff9c6048a70000
[ 8846.142661] R13: ffff9c5d09c50000 R14: ffff9c5d03b19000 R15: ffff9c5d09c502b0
[ 8846.142663] FS:  00007ab87b7a49c0(0000) GS:ffff9c605f580000(0000) knlGS:0000000000000000
[ 8846.142665] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 8846.142667] CR2: 0000782d86185000 CR3: 0000000166578002 CR4: 0000000000f72ef0
[ 8846.142669] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 8846.142671] DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
[ 8846.142673] PKRU: 55555554
[ 8846.142681] Call Trace:
[ 8846.142687]  <TASK>
[ 8846.142689]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.142864]  ? __warn.cold+0x93/0xed
[ 8846.142873]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.143044]  ? report_bug+0xe7/0x210
[ 8846.143048]  ? handle_bug+0x58/0x90
[ 8846.143054]  ? exc_invalid_op+0x19/0xc0
[ 8846.143058]  ? asm_exc_invalid_op+0x1a/0x20
[ 8846.143062]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.143227]  ? intel_fbc_activate+0x2e/0x90 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.143401]  intel_atomic_commit_tail+0x8e1/0xe70 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.143599]  intel_atomic_commit+0x33e/0x380 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.143795]  drm_atomic_commit+0xa9/0xe0
[ 8846.143802]  ? __pfx___drm_printfn_info+0x10/0x10
[ 8846.143806]  drm_atomic_connector_commit_dpms+0xe9/0x110
[ 8846.143811]  drm_mode_obj_set_property_ioctl+0x1b3/0x400
[ 8846.143816]  ? __pfx_drm_connector_property_set_ioctl+0x10/0x10
[ 8846.143819]  drm_connector_property_set_ioctl+0x3d/0x60
[ 8846.143823]  drm_ioctl_kernel+0xb0/0x100
[ 8846.143826]  drm_ioctl+0x277/0x4c0
[ 8846.143828]  ? __pfx_drm_connector_property_set_ioctl+0x10/0x10
[ 8846.143832]  __x64_sys_ioctl+0x94/0xd0
[ 8846.143837]  do_syscall_64+0x82/0x190
[ 8846.143841]  ? drm_ioctl_kernel+0xb0/0x100
[ 8846.143843]  ? __check_object_size+0x50/0x210
[ 8846.143847]  ? drm_ioctl+0x3c8/0x4c0
[ 8846.143849]  ? __pfx_drm_mode_destroyblob_ioctl+0x10/0x10
[ 8846.143855]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.143860]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.143864]  ? do_syscall_64+0x8e/0x190
[ 8846.143867]  ? switch_fpu_return+0x4e/0xd0
[ 8846.143871]  ? syscall_exit_to_user_mode+0x1d7/0x210
[ 8846.143874]  ? do_syscall_64+0x8e/0x190
[ 8846.143876]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.143879]  ? do_syscall_64+0x8e/0x190
[ 8846.143881]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.143885]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.143888]  ? do_syscall_64+0x8e/0x190
[ 8846.143890]  ? do_syscall_64+0x8e/0x190
[ 8846.143892]  ? syscall_exit_to_user_mode_prepare+0x148/0x170
[ 8846.143896]  ? syscall_exit_to_user_mode+0x10/0x210
[ 8846.143898]  ? do_syscall_64+0x8e/0x190
[ 8846.143901]  ? do_syscall_64+0x8e/0x190
[ 8846.143903]  ? do_syscall_64+0x8e/0x190
[ 8846.143905]  entry_SYSCALL_64_after_hwframe+0x76/0x7e
[ 8846.143908] RIP: 0033:0x7ab87c025ced
[ 8846.144010] Code: 04 25 28 00 00 00 48 89 45 c8 31 c0 48 8d 45 10 c7 45 b0 10 00 00 00 48 89 45 b8 48 8d 45 d0 48 89 45 c0 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 1a 48 8b 45 c8 64 48 2b 04 25 28 00 00 00
[ 8846.144013] RSP: 002b:00007ffcb019cd80 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 8846.144015] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007ab87c025ced
[ 8846.144017] RDX: 00007ffcb019ce10 RSI: 00000000c01064ab RDI: 0000000000000010
[ 8846.144019] RBP: 00007ffcb019cdd0 R08: 0000000000000002 R09: 0000000000000000
[ 8846.144021] R10: 00005804ebe1d7e0 R11: 0000000000000246 R12: 00007ffcb019ce10
[ 8846.144023] R13: 00000000c01064ab R14: 0000000000000010 R15: 00005804ebe230b0
[ 8846.144026]  </TASK>
[ 8846.144027] ---[ end trace 0000000000000000 ]---
[ 8846.892346] i915 0000:00:02.0: [drm] *ERROR* [CRTC:132:pipe B] mismatch in post_csc_lut hw_state doesn't match sw_state
[ 8846.892367] ------------[ cut here ]------------
[ 8846.892369] i915 0000:00:02.0: [drm] pipe state doesn't match!
[ 8846.892555] WARNING: CPU: 8 PID: 2514 at /var/lib/dkms/i915-sriov-dkms/2024.12.02/build/drivers/gpu/drm/i915/display/intel_modeset_verify.c:222 intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.892866] Modules linked in: tls hid_multitouch nft_masq nft_ct xt_conntrack xt_MASQUERADE nft_reject_ipv4 nf_conntrack_netlink nf_reject_ipv4 nft_reject xfrm_user xfrm_algo act_csum cls_u32 sch_htb xt_addrtype nft_compat nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 scsi_transport_iscsi overlay cmac algif_hash algif_skcipher af_alg nf_tables usbip_host pkcs8_key_parser snd_seq_dummy snd_seq_oss snd_seq_midi_event snd_seq cuse elan_i2c ec_sys v4l2loopback(OE) kvmgt mdev vfio_iommu_type1 vfio iommufd vhost_net vhost vhost_iotlb tap acpi_power_meter uvcvideo videobuf2_vmalloc uvc videobuf2_memops videobuf2_v4l2 videobuf2_common videodev usbhid sunrpc acpi_ipmi virtio_mmio virtio_scsi virtio_rng 9pnet_virtio 9pnet netfs virtio_console virtio_net net_failover failover kvm_intel kvm ipmi_watchdog ipmi_devintf ipmi_msghandler ums_cypress lis3lv02d_i2c lis3lv02d btsdio btusb btmtk firewire_ohci firewire_core crc_itu_t dcdbas dell_wmi_descriptor dm_zero coretemp mspro_block tifm_ms memstick tifm_sd
[ 8846.892952]  mmc_core tifm_core snd_soc_skl_hda_dsp uio_cif snd_soc_intel_sof_board_helpers uio snd_soc_intel_hda_dsp_common hidp snd_sof_probes bnep vfat tun fat snd_hda_codec_hdmi rfcomm snd_hda_codec_conexant snd_hda_codec_generic snd_soc_dmic snd_usb_audio snd_usbmidi_lib snd_ump snd_sof_pci_intel_mtl snd_rawmidi snd_sof_intel_hda_generic snd_seq_device soundwire_intel mc soundwire_cadence snd_sof_intel_hda_common hci_uart snd_soc_hdac_hda btqca pwrseq_core snd_sof_intel_hda_mlink btrtl snd_sof_intel_hda btintel snd_sof_pci snd_sof_xtensa_dsp btbcm snd_sof snd_sof_utils snd_hda_ext_core snd_soc_acpi_intel_match soundwire_generic_allocation snd_soc_acpi soundwire_bus snd_soc_core snd_compress ac97_bus snd_pcm_dmaengine snd_hda_intel processor_thermal_device_pci mousedev bluetooth processor_thermal_device snd_intel_dspcfg psmouse serio_raw processor_thermal_wt_hint snd_intel_sdw_acpi mei_gsc_proxy spd5118 spi_nor processor_thermal_rfim atkbd think_lmi snd_hda_codec libps2 intel_rapl_msr vivaldi_fmap snd_ctl_led mtd
[ 8846.893012]  firmware_attributes_class mei_me i2c_i801 snd_hda_core snd_pcm_oss processor_thermal_rapl intel_lpss_pci i2c_smbus wmi_bmof snd_hwdep snd_mixer_oss ucsi_acpi intel_rapl_common intel_lpss i2c_mux processor_thermal_wt_req mei typec_ucsi thunderbolt snd_pcm processor_thermal_power_floor thinkpad_acpi intel_vpu idma64 typec processor_thermal_mbox igen6_edac snd_timer platform_profile roles i2c_hid_acpi snd joydev intel_pmc_core soundcore i2c_hid int3403_thermal yenta_socket int340x_thermal_zone pcmcia_rsrc intel_vsec i8042 intel_hid pcmcia_core pmt_telemetry int3400_thermal acpi_tad serio mac_hid pmt_class pinctrl_meteorlake sparse_keymap acpi_pad acpi_thermal_rel iwlmvm mac80211 e1000e zram libarc4 842_decompress ptp 842_compress lz4hc_compress pps_core lz4_compress hid_generic iwlwifi cfg80211 rfkill uas usb_storage exfat vhci_hcd usbip_core serpent_avx_x86_64 twofish_avx_x86_64 twofish_x86_64_3way twofish_x86_64 twofish_generic twofish_common serpent_generic ccm xts ntfs3 br_netfilter bridge stp llc uinput
[ 8846.893083]  dell_smm_hwmon i2c_dev sg vhba(OE) crypto_user acpi_call(OE) loop nfnetlink ip_tables x_tables dm_crypt cbc encrypted_keys trusted asn1_encoder tee crct10dif_pclmul crc32_pclmul polyval_clmulni polyval_generic ghash_clmulni_intel sha512_ssse3 sha256_ssse3 sha1_ssse3 aesni_intel gf128mul nvme crypto_simd cryptd spi_intel_pci nvme_core spi_intel nvme_auth raid1 dm_raid raid456 libcrc32c async_raid6_recov async_memcpy async_pq async_xor xor async_tx raid6_pq dm_mod md_mod i915(OE) drm_buddy drm_display_helper ttm video wmi i2c_algo_bit cec intel_gtt ext4 crc32c_generic crc32c_intel crc16 mbcache jbd2 [last unloaded: hid_multitouch]
[ 8846.893136] CPU: 8 UID: 0 PID: 2514 Comm: Xorg Tainted: G     U  W  OE      6.12.4-arch1-1.1 #1 606ae73903c461ea514adc42e53e980d964fe09f
[ 8846.893144] Tainted: [U]=USER, [W]=WARN, [O]=OOT_MODULE, [E]=UNSIGNED_MODULE
[ 8846.893147] Hardware name: LENOVO 21M7002VPB/21M7002VPB, BIOS R2JET35W(1.12 ) 07/05/2024
[ 8846.893150] RIP: 0010:intel_modeset_verify_crtc+0x2c6/0x4b0 [i915]
[ 8846.893353] Code: b3 0b 00 49 8b 7d 08 48 8b 5f 50 48 85 db 75 03 48 8b 1f e8 5c ef bb c1 48 89 da 48 c7 c7 18 d1 bc c0 48 89 c6 e8 aa 36 13 c1 <0f> 0b 31 f6 4c 89 e7 48 c7 c2 e3 24 bf c0 e8 27 2a fb ff 48 c7 c2
[ 8846.893356] RSP: 0018:ffffae6b013e3760 EFLAGS: 00010282
[ 8846.893360] RAX: 0000000000000000 RBX: ffff9c5d025467a0 RCX: 0000000000000027
[ 8846.893363] RDX: ffff9c605f8218c8 RSI: 0000000000000001 RDI: ffff9c605f8218c0
[ 8846.893365] RBP: ffff9c5f007e0000 R08: 00000000ffff7fff R09: ffff9c606fe3ffa8
[ 8846.893367] R10: 00000000ffff7fff R11: 0000000000000003 R12: ffff9c5f007e2000
[ 8846.893369] R13: ffff9c5d09c50000 R14: ffff9c5d03b19000 R15: ffff9c5d09c502b0
[ 8846.893372] FS:  00007ab87b7a49c0(0000) GS:ffff9c605f800000(0000) knlGS:0000000000000000
[ 8846.893375] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 8846.893377] CR2: 000058ba2bd869a0 CR3: 0000000166578005 CR4: 0000000000f72ef0
[ 8846.893379] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 8846.893381] DR3: 0000000000000000 DR6: 00000000ffff07f0 DR7: 0000000000000400
[ 8846.893383] PKRU: 55555554
[ 8846.893385] Call Trace:
[ 8846.893390]  <TASK>
[ 8846.893392]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.893587]  ? __warn.cold+0x93/0xed
[ 8846.893598]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.893801]  ? report_bug+0xe7/0x210
[ 8846.893806]  ? handle_bug+0x58/0x90
[ 8846.893813]  ? exc_invalid_op+0x19/0xc0
[ 8846.893817]  ? asm_exc_invalid_op+0x1a/0x20
[ 8846.893822]  ? intel_modeset_verify_crtc+0x2c6/0x4b0 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.894024]  ? intel_fbc_activate+0x2e/0x90 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.894234]  intel_atomic_commit_tail+0x8e1/0xe70 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.894446]  intel_atomic_commit+0x33e/0x380 [i915 cff4e35cc491b3558b5f0361118eb47e271db17e]
[ 8846.894650]  drm_atomic_commit+0xa9/0xe0
[ 8846.894656]  ? __pfx___drm_printfn_info+0x10/0x10
[ 8846.894661]  drm_atomic_connector_commit_dpms+0xe9/0x110
[ 8846.894666]  drm_mode_obj_set_property_ioctl+0x1b3/0x400
[ 8846.894672]  ? __pfx_drm_connector_property_set_ioctl+0x10/0x10
[ 8846.894675]  drm_connector_property_set_ioctl+0x3d/0x60
[ 8846.894678]  drm_ioctl_kernel+0xb0/0x100
[ 8846.894682]  drm_ioctl+0x277/0x4c0
[ 8846.894685]  ? __pfx_drm_connector_property_set_ioctl+0x10/0x10
[ 8846.894689]  __x64_sys_ioctl+0x94/0xd0
[ 8846.894694]  do_syscall_64+0x82/0x190
[ 8846.894698]  ? update_load_avg.constprop.0+0x79/0x620
[ 8846.894702]  ? sched_clock+0x10/0x30
[ 8846.894706]  ? psi_group_change+0x187/0x3c0
[ 8846.894710]  ? psi_task_switch+0xe0/0x2d0
[ 8846.894714]  ? finish_task_switch.isra.0+0x99/0x2e0
[ 8846.894719]  ? __schedule+0x421/0x21e0
[ 8846.894722]  ? __rseq_handle_notify_resume+0x23f/0x510
[ 8846.894727]  ? syscall_exit_to_user_mode+0x1d7/0x210
[ 8846.894731]  ? do_syscall_64+0x8e/0x190
[ 8846.894734]  ? task_tick_fair+0x3a5/0x1110
[ 8846.894737]  ? sched_use_asym_prio+0x67/0x90
[ 8846.894739]  ? sched_balance_trigger+0x14a/0x380
[ 8846.894743]  ? update_process_times+0xb6/0xd0
[ 8846.894748]  ? timerqueue_add+0xa3/0xc0
[ 8846.894751]  ? enqueue_hrtimer+0x38/0x90
[ 8846.894755]  ? __hrtimer_run_queues+0x16b/0x2a0
[ 8846.894759]  ? rcu_core+0x1a3/0x390
[ 8846.894762]  ? sched_clock+0x10/0x30
[ 8846.894765]  ? sched_clock_cpu+0xb/0x30
[ 8846.894768]  ? irqtime_account_irq+0x3e/0xc0
[ 8846.894773]  ? handle_softirqs+0x1a0/0x2b0
[ 8846.894778]  ? sched_clock+0x10/0x30
[ 8846.894781]  ? irq_exit_rcu+0x53/0xc0
[ 8846.894784]  entry_SYSCALL_64_after_hwframe+0x76/0x7e
[ 8846.894788] RIP: 0033:0x7ab87c025ced
[ 8846.894887] Code: 04 25 28 00 00 00 48 89 45 c8 31 c0 48 8d 45 10 c7 45 b0 10 00 00 00 48 89 45 b8 48 8d 45 d0 48 89 45 c0 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 1a 48 8b 45 c8 64 48 2b 04 25 28 00 00 00
[ 8846.894889] RSP: 002b:00007ffcb019cef0 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 8846.894892] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007ab87c025ced
[ 8846.894894] RDX: 00007ffcb019cf80 RSI: 00000000c01064ab RDI: 0000000000000010
[ 8846.894896] RBP: 00007ffcb019cf40 R08: 0000000000000002 R09: 0000000000000000
[ 8846.894897] R10: 00005804ec7d49f0 R11: 0000000000000246 R12: 00007ffcb019cf80
[ 8846.894899] R13: 00000000c01064ab R14: 0000000000000010 R15: 00005804c1566980
[ 8846.894902]  </TASK>
[ 8846.894903] ---[ end trace 0000000000000000 ]---
`

**Comment by bbaa-bbaa** [comment]
Because Intel's upstream tree includes DRM subsystem modifications related to the Degamma LUT that have not been submitted to the mainline kernel, I couldn't make this module work without porting the DRM module. I just simply reverted the relevant patch. There might be some side effects, and it will take a few days to investigate. 

It seems that https://github.com/moetayuko/intel-gpu-i915-backport has implemented a proper porting. Are you interested in trying it out?

**Comment by makoONE** [comment]
I have tested the new port with kernel 6.11.0-2-pve and 6.8.12-5-pve. Nice that it works with 6.8.12-5-pve but unfortunately the behavior is unchanged. The error “*ERROR* [CRTC:81:pipe A] mismatch in post_csc_lut hw_state doesn't match sw_state” still appears and the windows with the console applications still freeze during resizing.

**Comment by bbaa-bbaa** [comment]
Which CPU  model are you using?
Can you try https://github.com/moetayuko/intel-gpu-i915-backports?

**Comment by patrykk** [comment]
Intel(R) Core(TM) Ultra 7 155H

**Comment by bbaa-bbaa** [comment]
> Intel(R) Core(TM) Ultra 7 155H

Can you try https://github.com/moetayuko/intel-gpu-i915-backports？
Just use the deb file on the releases page.

**Comment by patrykk** [comment]
I use arch linux so I need to recompile it with kernel 6.12.4-arch1-1.1 ( its a kernel from x86-64-v3 repo )

**Comment by makoONE** [comment]
> > Intel(R) Core(TM) Ultra 7 155H
> 
> Can you try [https://github.com/moetayuko/intel-gpu-i915-backports？](https://github.com/moetayuko/intel-gpu-i915-backports%EF%BC%9F) Just use the deb file on the releases page.

I have tested the driver. At first everything works fine, but after a while the entire system freezes, triggered by a kernel NULL pointer dereference error.

My CPU is a 12th Gen Intel(R) Core(TM) i7-12650H.

**Comment by bbaa-bbaa** [comment]
It seems that this issue only occurs on mobile platforms. I’m unable to reproduce it with my 12th Gen Intel(R) Core(TM) i5-12400.

After #11, this repository has been synchronized with the 6.6/linux branch of linux-intel-lts, so it’s likely an issue in the upstream repository.

The next step might be switching the upstream to the linux/6.8 branch of https://github.com/intel/mainline-tracking, but it will take some time.

**Comment by bbaa-bbaa** [comment]
@patrykk Can you try this kernel built from the mainline-tracking linux/6.6 branch? The SR-IOV feature is already integrated into the i915 driver in the kernel, so you need to uninstall the currently installed i915 DKMS module before installing this kernel.

If this branch doesn’t have the issue, I’ll attempt to backport it in few weeks.
Thanks.

Prebulit kernel can install with: `pacman -U https://bbaa.moe/linux-intel-mainline-6.8.0.intel1-1-x86_64.pkg.tar.zst`
kernel header: `pacman -U https://bbaa.moe/linux-intel-mainline-headers-6.8.0.intel1-1-x86_64.pkg.tar.zst`

[PKGBUILD.zip](https://github.com/user-attachments/files/18136676/PKGBUILD.zip)

**Comment by bbaa-bbaa** [comment]
@makoONE Can you try this kernel built from the mainline-tracking linux/6.6 branch? The SR-IOV support is already integrated into the i915 driver in the kernel.

You can install with dpkg like:
```bash
wget https://bbaa.moe/proxmox-kernel-6.8.1-99-pve_6.8.1-1_amd64.deb
dpkg -i proxmox-kernel-6.8.1-99-pve_6.8.1-1_amd64.deb
```
and choose the kernel in the boot menu
![Boot Menu](https://github.com/user-attachments/assets/40d5880c-ae0f-4811-8482-8aa46c3c77c6)
![Select Kernel](https://github.com/user-attachments/assets/5a358710-d838-4269-bf08-b25a7fa9f8ba)

You can uninstall it with `dpkg -r proxmox-kernel-6.8.1-99-pve`

Thanks.

**Comment by makoONE** [comment]
@bbaa-bbaa Thanks for your efforts, I have tested your kernel. My Proxmox server initially starts well in the X environment, but as soon as a VM with GPU access via SR-IOV starts, the system becomes unbearably slow and thus unusable.
Dozens of dmeg error messages appear in this context:
[  136.116274] i915 0000:00:02.0: [drm] *ERROR* GT0: GUC: TLB invalidation response timed out for seqno 171
[  138.164102] i915 0000:00:02.0: [drm] *ERROR* GT0: GUC: TLB invalidation response timed out for seqno 172
[  138.164105] i915 0000:00:02.0: [drm] *ERROR* GT0: GUC: TLB invalidation response timed out for seqno 173
....

**Comment by bbaa-bbaa** [comment]
@makoONE Thanks for your test. Perhaps we need to do more tests to find the upstream branch that works correctly.

Can you try this kernel built from the mainline-tracking linux/6.11 branch?
You can install with dpkg like:
```bash
wget https://bbaa.moe/proxmox-kernel-6.11.0-1-pve-intel_6.11.0-2_amd64.deb
dpkg -i proxmox-kernel-6.11.0-1-pve-intel_6.11.0-2_amd64.deb
```

and choose the kernel in the boot menu
![image](https://github.com/user-attachments/assets/40d5880c-ae0f-4811-8482-8aa46c3c77c6)
![image](https://github.com/user-attachments/assets/070b8e75-a089-4f69-8962-09a265638224)

You can uninstall it with `dkms -r proxmox-kernel-6.11.0-1-pve-intel` after testing;

**Comment by makoONE** [comment]
@bbaa-bbaa I was happy to test the new kernel. I think it works the same as with your 2024.12.10 dkms build, except that fortunately there are no more error messages ‘*ERROR* [CRTC:81:pipe A] mismatch in post_csc_lut hw_state doesn't match sw_state’ in the dmesg.
The local X environment on the Proxmox server runs flawlessly with native X applications.
A VM running on the same server with GPU sharing via SR-IOV runs without errors and at normal speed.
The obscure error remains in the local X GUI with the text-based apps in the window such as mc (/usr/bin/x-terminal-emulator -e mc) and htop (/usr/bin/x-terminal-emulator -e htop).
Whenever I drag the edge of the window with the mouse to enlarge it, the entire desktop freezes. However, the VM continues to run undisturbed and I can work on it from outside via RDP.
There are no error messages in the logs when it freezes, and starting the problematic apps with the verbose option does not give any indication either.

All in all, a very good result so far and thank you for that.

**Comment by bbaa-bbaa** [comment]
Would this issue occur when using the original kernel (without installing any dkms module)?

**Comment by makoONE** [comment]
Oh yes, I forgot to mention that. No the error does not occur with any unmodified pve kernel.

**Comment by makoONE** [comment]
Tested with kernel 6.11.0-1-pve-intel - The error no longer occurs after the setting (sriov_numvfs=0).

**Comment by bbaa-bbaa** [comment]
try https://github.com/bbaa-bbaa/i915-sriov-dkms/tree/post_csc_lut

```shell
git clone https://github.com/bbaa-bbaa/i915-sriov-dkms.git
cd i915-sriov-dkms
git checkout post_csc_lut
```

**Comment by makoONE** [comment]
For me, the new build makes no difference to the previous one. The error ‘*ERROR* [CRTC:81:pipe A] mismatch in post_csc_lut hw_state doesn't match sw_state’ still appears in the dmesg and resizing the windows of apps like mc and htop still causes the whole desktop to freeze.

**Comment by bbaa-bbaa** [comment]
Try https://github.com/bbaa-bbaa/i915-sriov-dkms/tree/kernel-v6.12-backport, base on https://github.com/intel/mainline-tracking linux/6.12 branch.
Only support 6.11-6.12 kernel.

You can checkout branch with
```shell
git clone https://github.com/bbaa-bbaa/i915-sriov-dkms.git
git checkout kernel-v6.12-backport
```

**Comment by makoONE** [comment]
The new dkms build, built under kernel 6.11.0-2-pve, behaves exactly the same in my tests as with the Intel kernel (proxmox-kernel-6.11.0-1-pve-intel) before.

https://github.com/bbaa-bbaa/i915-sriov-dkms/issues/10#issuecomment-2549319250

**Comment by bbaa-bbaa** [comment]
Glad to see that the issue with post_csc_lut mismatch has been fixed. The problem causing the desktop freeze seems to be another issue related to SR-IOV. Would you be interested in opening a new issue to track the problem?
